### ISSUE TITLE
Query Regarding usage of "time_varying_unknown" variables

### ISSUE THREAD

**Issue by Divtripa** [jdb78/pytorch-forecasting#522]
I have been using Pytorch forecasting with a dataset which has many covariates. The performance on the validation set was excellent, however the performance on the "new_prediction_data" as shown in the tutorial notebook (https://pytorch-forecasting.readthedocs.io/en/latest/tutorials/stallion.html) was not upto the mark. I used all the static covariates as "known" (and assigned them in the new_prediction_data accordingly, The future data was available). However, for the unknown variables I had to use either an aggregate (mean / mode) or the latest value. Does the decoder data have to be filled in with their "future" values as well? Are the "time-varying unknowns"  actively used in the decoding process as well?

As far as I've read the TFT paper it isn't the case but just wanted to confirm. If so, the target and the log of the target are also present in that category (in the tutorials) and wouldn't it mean that the prediction would be highly compromised?

Could you please provide of what would be **the ideal values** for the time varying unknowns? known future values, aggregates, latest values or any random sample?

Just to add, the library works very well and performs great much better than GluonTS even though it has a higher level of abstraction. Thanks a lot for the effort you've put in.

**Comment by Sunishchal** [comment]
@Divtripa did you find the answers to your questions? If so, please do share. 

I'm specifically wondering how one might feed "lead" or forward-looking features such as your baseline forecast from an existing demand planning system. My understanding is sequence models do not allow these features to be in a "wide" form (baseline_lead_1, baseline_lead_2, etc.) However, feeding it in long form would now allow you to provide an updated forward looking snapshot at the next time_idx without overwriting the previous snapshot's values. 

@jdb78 if you're able to shed some insight, I'd appreciate it.

**Comment by Divtripa** [comment]
Hey @Sunishchal,
Regarding the answer to my question: in my experience Decoder-data can be filled with any dummy value for future_unknown_categorical / real values. For the forward looking value they should be filled with the future values. Time-varying Unknowns are not used in the decoder layers so their values can be anything.

Regarding your requirements: I think you can just pass the baseline forecasts as a future_unknown_real feature. You could map the values to the dates (in the long-format as you mentioned and maybe use a default value such as 0 or -1 for the past, and the forward-looking values for the future). The target will have to be in the future_unknown list, and this can be a new column in the known_real class.
